core/src/main/java/org/wso2/msf4j/internal/websocket/EndpointDispatcher.java

```diff
@@ -0,0 +1,165 @@
+     * @throws WebSocketEndpointAnnotationException throws if {@link ServerEndpoint} is not found.
+     */
+    public String getUri(Object webSocketEndpoint) throws WebSocketEndpointAnnotationException {
+        ServerEndpoint serverEndpointAnnotation = webSocketEndpoint.getClass().getAnnotation(ServerEndpoint.class);
+        if (serverEndpointAnnotation == null) {
+            throw new WebSocketEndpointAnnotationException("ServerEndpoint is not declared");
+        }
+        return serverEndpointAnnotation.value();
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
+     * @return method to handle new connection if exists else null.
+     */
+    public Method getOnOpenMethod(Object webSocketEndpoint) {
+        Method[] methods = webSocketEndpoint.getClass().getMethods();
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(OnOpen.class)) {
+                return method;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
+     * @return method to handle connection closure if exists else null.
+     */
+    public Method getOnCloseMethod(Object webSocketEndpoint) {
+        Method[] methods = webSocketEndpoint.getClass().getMethods();
+        if (methods == null) {
+            return null;
+        }
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(OnClose.class)) {
+                return method;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
+     * @return method to handle errors if exists else null.
+     */
+    public Method getOnErrorMethod(Object webSocketEndpoint) {
+        Method[] methods = webSocketEndpoint.getClass().getMethods();
+        if (methods == null) {
+            return null;
+        }
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(OnError.class)) {
+                return method;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
+     * @return method to handle String messages if exists else null.
+     */
+    public Method getOnStringMessageMethod(Object webSocketEndpoint) {
+        Method[] methods = webSocketEndpoint.getClass().getMethods();
+        if (methods == null) {
+            return null;
+        }
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(OnMessage.class)) {
+                //Adding OnMessage according to their types
+                Class<?>[] paraTypes = method.getParameterTypes();
+                List<Class<?>> paraList = Arrays.asList(paraTypes);
+                if (paraList.contains(String.class)) {
+                    return method;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
```

Comment:
Method description is missing. Explain what the method does in the first line ending with a full stop